src/mpid/ch4/src/ch4_init.c

```diff
@@ -102,6 +102,18 @@
       description : >-
         Defines the threshold of high-density datatype. The
         density is calculated by (datatype_size / datatype_num_contig_blocks).
+
+    - name        : MPIR_CVAR_CH4_THREAD_ID_KEY
+      category    : CH4
+      type        : string
+      default     : "thread_id"
+      class       : device
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_ALL_EQ
+      description : >-
+        This CVAR specifies the key for a comm info hint that can define thread ID for a communicator
+        or to control the mapping a communicator with a tx and rcv vci.
+
```

Comment:
> This feature can be useful to certain applications, e.g., OneCCL.

Could you elaborate on the use case?

Reply:
The initial motivation for this hint was to allow an application to smoothly switch from using IMPI to MPICH, with minimal changes. IMPI provides a similar feature to dynamically rename the key. An application can use different keys, e.g., "thread_id", "ep_idx", "comm_id", in different contexts. Or this value can change across different applications. When MPICH allows dynamic renaming, an application can run with either IMPI or MPICH without having to change the key string in the source code.

Reply:
I see. In stead of adding this CVAR, which is cumbersome for user to use anyway, let's make `thread_id`, `ep_idx`, `comm_id` aliases, i.e. check against all aliases in `comm_hint_test`.

Reply:
> the aliases can by changed freely by the applications.

Why would applications do that?

Reply:
Hints itself is a hack (until official API), adding more complexity to hints is over-engineering in my opinion. I think just offer a few aliases (only when there are real usages) is sufficient.

> So MPI will have 2 users – mpi4py and CCL, so flexible hint names may be useful here.

What's the issue with 2 users? They can use the same hint key, right?

Reply:
Here are some updates after offline discussions: We can get rid of the CVAR - CCL should now be able to make make changes and adapt to MPICH. If needed, we could support multiple hard-coded aliases for the hint instead. However, at present, a single name for this hint should be sufficient.

Reply:
Removed the CVAR and changed the name "thread_id" to "vci".